### PR TITLE
ESQL: fix the column position in errors

### PR DESCRIPTION
**Pull request by bpintea**
This fixes the off-by-one error of the column position in some of the error messages.

### CODE REVIEW

File: docs/reference/esql/functions/kibana/definition/count.json
Comment (bpintea) [inline docs/reference/esql/functions/kibana/definition/count.json:1]: The JSONs in this PR aren't directly related, they just haven't been checked in along their corresponding PRs.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ParsingException.java [hunk 1]
```diff
@@ -28,13 +28,13 @@ public ParsingException(String message, Exception cause, int line, int charPosit
     public ParsingException(Source source, String message, Object... args) {
         super(message, args);
         this.line = source.source().getLineNumber();
-        this.charPositionInLine = source.source().getColumnNumber();
+        this.charPositionInLine = source.source().getColumnNumber() - 1;
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ParsingException.java:31]: `Location#getColumnNumber()` increments the count by 1 already, which this class also does in its own `getColumnNumber()` -- the bug.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java [hunk 2]
```diff
@@ -511,7 +511,7 @@ private static void checkRow(LogicalPlan p, Set<Failure> failures) {
         if (p instanceof Row row) {
             row.fields().forEach(a -> {
                 if (DataType.isRepresentable(a.dataType()) == false) {
-                    failures.add(fail(a, "cannot use [{}] directly in a row assignment", a.child().sourceText()));
+                    failures.add(fail(a.child(), "cannot use [{}] directly in a row assignment", a.child().sourceText()));
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java:514]: Related, though not caused by the off-by-one error: we need to provide the location of the query excerpt we provide the user (as we do everywhere else).

**Comment by bpintea** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ParsingException.java (hunk 1 above)
Comment (fang-xing-esql) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ParsingException.java:31]: This is interesting find, we are really off by one, I have never noticed it 😂 ! It worths a comment.

I wonder if it is a good idea to replace the three math with one here, and keep the column number from source unchanged? 

```
    public ParsingException(String message, Exception cause, int line, int charPositionInLine) {
        super(message, cause);
        this.line = line;
        this.charPositionInLine = charPositionInLine + 1;
    }

    ParsingException(String message, Object... args) {
        this(Source.EMPTY, message, args);
    }

    public ParsingException(Source source, String message, Object... args) {
        super(message, args);
        this.line = source.source().getLineNumber();
        this.charPositionInLine = source.source().getColumnNumber();
    }

    public ParsingException(Exception cause, Source source, String message, Object... args) {
        super(cause, message, args);
        this.line = source.source().getLineNumber();
        this.charPositionInLine = source.source().getColumnNumber();
    }

    public int getLineNumber() {
        return line;
    }

    public int getColumnNumber() {
        return charPositionInLine;
    }
```

**Review by fang-xing-esql — APPROVED** [review]
Thank you @bpintea, LGTM! I added a comment, but you can decide whether to keep it as it or not.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ParsingException.java (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ParsingException.java]: Indeed, thanks!

**Pull request merged**